File: Week 2/5. Binary Sequence Generation/binary_sequence_generation.py
def generateBinarySequence(size, sequence, index):
    if (index == size):
        print(''.join(map(str, sequence)))
        return
    
    sequence[index] = 0
    generateBinarySequence(size, sequence, index + 1)
    
    sequence[index] = 1 
    generateBinarySequence(size, sequence, index + 1)
    
size = int(input())
sequence = [0] * size
generateBinarySequence(size, sequence, 0)

File: Week 2/5. Binary Sequence Generation/BinarySequenceGeneration.java
//JAVA 
import java.util.*; 

public class BinarySequenceGeneration { 
    public static void main(String[] args) { 
        Scanner input = new Scanner(System.in);
        int size = input.nextInt();
        
        int[] sequence = new int[size];
        
        generateBinarySequence(size, sequence, 0);
        input.close();
    }
    
    public static void generateBinarySequence(int size, int[] sequence, int index) {
        if (index == size) {
            for (int num : sequence) {
                System.out.print(num);
            }
            System.out.println();
            return;
        }
        
        sequence[index] = 0;
        generateBinarySequence(size, sequence, index + 1);
        
        sequence[index] = 1;
        generateBinarySequence(size, sequence, index + 1);
    }
}

